### PR TITLE
Add tests for generated plots of `matplotlib.plot_edf`

### PR DESCRIPTION
**Pull request by makinzm**
## Motivation

Add test in matplotlib test ``plot_edf``  of matplotlib.

## Description of the changes

This test confirms 
- that the values are monotonically non-decreasing and  
- that the values are in [0,1].
## Reference
- [Stabilise visualisation module #2959](https://github.com/optuna/optuna/issues/2959)  
- [Add tests for generated plots of plot_edf #3188](https://github.com/optuna/optuna/pull/3188)

### CODE REVIEW

**Review by HideakiImamura — APPROVED** [review]
Thanks for the PR! LGTM!

File: tests/visualization_tests/matplotlib_tests/test_edf.py [hunk 1]
```diff
@@ -11,6 +14,14 @@ def test_target_is_none_and_study_is_multi_obj() -> None:
         plot_edf(study)
 
 
+def _validate(y_values: List[float]) -> None:
```
Comment (toshihikoyanase) [inline tests/visualization_tests/matplotlib_tests/test_edf.py:17]: Maybe `Sequence` is preferrable if we can use it.
```suggestion
def _validate(y_values: Sequence[float]) -> None:
```

Comment (makinzm) [reply tests/visualization_tests/matplotlib_tests/test_edf.py:17]: Thank you for your advice. I will commit after modifying and checking.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your quick fix. LGTM!

**Pull request merged**